srcpkgs/widelands/patches/wordsize.patch

```diff
@@ -0,0 +1,23 @@
+--- a/src/base/macros.h
++++ b/src/base/macros.h
+@@ -28,6 +28,11 @@
+ #endif
+ #endif
+
++/* Used for wordsize only */
++#if !defined(_WIN32) && !defined(_WIN64)
++#include <limits.h>
++#endif
++
+ /* Macros for disabling GCC warnings and errors
+  * From http://dbp-consulting.com/tutorials/SuppressingGCCWarnings.html and
+  * slightly modified to remove support entirely for GCC < 4.6 because we'll
+@@ -135,7 +140,7 @@
+ #define PRIuS PRIu32
+ #endif
+ #else
+-#if __WORDSIZE == 64
++#if (ULONG_MAX == 0xffffffffffffffff)
+ #define PRIuS "lu"
+ #else
```

Comment:
You need to have a patch that highlight 32bit platform, too.

Reply:
Not marking as resolved yet. I hope `#if (ULONG_MAX == 0xffffffff)` is the right directive for 32 bit platforms.

Reply:
Hm, it can be simple than that, it's `zu`